In [1]:
library("MASS")

## Problema 1
### Solución

In [2]:

#Imprimimos la tabla con los datos 

datosAccidentes = array(data=c(7287,10381,3246,6123,11587,10969,6134,6693
         ,175,136,73,141,126,83,94,74,720,566,710,710,
         577,259,564,353,91,96,159,188,48,37,82,74,10,14,
        31,45,8,1,17,12),
dim = c(2,2,2,5), 
dimnames = list("genero" = c("femenino","masculino"),
"localizacion" = c("rural","urbano"),
"cinturon" = c("si","no"),
"resultado"=c( "1" ,"2" , "3", "4", "5")))

dataTable=ftable(datosAccidentes,
        row.vars=c("genero","localizacion","cinturon"))
#convertimos a data frame para aplicar los analisis
accidentes<- as.data.frame(as.table(datosAccidentes))
#imprimimos la tabla
dataTable
accidentes[1:10,]



                                resultado     1     2     3     4     5
genero    localizacion cinturon                                        
femenino  rural        si                  7287   175   720    91    10
                       no                 11587   126   577    48     8
          urbano       si                  3246    73   710   159    31
                       no                  6134    94   564    82    17
masculino rural        si                 10381   136   566    96    14
                       no                 10969    83   259    37     1
          urbano       si                  6123   141   710   188    45
                       no                  6693    74   353    74    12

genero    localizacion cinturon resultado Freq 
1  femenino  rural        si       1          7287
2  masculino rural        si       1         10381
3  femenino  urbano       si       1          3246
4  masculino urbano       si       1          6123
5  femenino  rural        no       1         11587
6  masculino rural        no       1         10969
7  femenino  urbano       no       1          6134
8  masculino urbano       no       1          6693
9  femenino  rural        si       2           175
10 masculino rural        si       2           136

In [3]:
print("calculamos los valores marginales totales")
addmargins(dataTable)

[1] "calculamos los valores marginales totales"


Sum  
     7287 175  720  91  10  8283
    11587 126  577  48   8 12346
     3246  73  710 159  31  4219
     6134  94  564  82  17  6891
    10381 136  566  96  14 11193
    10969  83  259  37   1 11349
     6123 141  710 188  45  7207
     6693  74  353  74  12  7206
Sum 62420 902 4459 775 138 68694

In [4]:
print("calculamos la proporcion para cada celda")
prop.table(dataTable)[,1:5]

[1] "calculamos la proporcion para cada celda"


[,1]       [,2]        [,3]        [,4]         [,5]        
[1,] 0.10607913 0.002547530 0.010481265 0.0013247154 1.455731e-04
[2,] 0.16867558 0.001834221 0.008399569 0.0006987510 1.164585e-04
[3,] 0.04725304 0.001062684 0.010335692 0.0023146126 4.512767e-04
[4,] 0.08929455 0.001368387 0.008210324 0.0011936996 2.474743e-04
[5,] 0.15111946 0.001979794 0.008239439 0.0013975020 2.038024e-04
[6,] 0.15967916 0.001208257 0.003770344 0.0005386205 1.455731e-05
[7,] 0.08913442 0.002052581 0.010335692 0.0027367747 6.550790e-04
[8,] 0.09743209 0.001077241 0.005138731 0.0010772411 1.746877e-04

### Modelo de independencia

Ahora vamos a modelar los datos con el modelo de independencia, diremos que el conteo por celda
con las catégorias $i$, $j$, $k$, $l$ es igual al producto de las densidades marginales y el número de 
observaciones. $\mu_{ij}=n\pi_i\pi_j\pi_k\pi_l$
Asi tendremos que:
\begin{eqnarray}
\log(\mu_{ijkl})=\log n+\log \pi_i+\log \pi_j+\log \pi_k+\log \pi_l
\end{eqnarray}

\begin{eqnarray}
\log(\mu_{ijkl})=\lambda+\lambda_i^X+\lambda_j^Y+\lambda_k^W+\lambda_l^Z
\end{eqnarray}



In [5]:
mod0 <- glm(Freq ~ genero + localizacion + cinturon+ resultado, 
            data = accidentes, family = poisson)
summary(mod0)
cbind(mod0$data, round(fitted(mod0),digits = 0) )[1:10,]


Call:
glm(formula = Freq ~ genero + localizacion + cinturon + resultado, 
    family = poisson, data = accidentes)

Deviance Residuals: 
     Min        1Q    Median        3Q       Max  
-24.1167   -4.6372   -0.9405    5.6246   17.2114  

Coefficients:
                    Estimate Std. Error z value Pr(>|z|)    
(Intercept)         9.006192   0.007706 1168.75   <2e-16 ***
generomasculino     0.152155   0.007653   19.88   <2e-16 ***
localizacionurbano -0.525589   0.007896  -66.56   <2e-16 ***
cinturonno          0.201277   0.007669   26.24   <2e-16 ***
resultado2         -4.237026   0.033536 -126.34   <2e-16 ***
resultado3         -2.638961   0.015501 -170.24   <2e-16 ***
resultado4         -4.388778   0.036143 -121.43   <2e-16 ***
resultado5         -6.114387   0.085219  -71.75   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 176952  on 39  degrees of freedom
Residual deviance:

genero    localizacion cinturon resultado Freq 
1  femenino  rural        si       1          7287
2  masculino rural        si       1         10381
3  femenino  urbano       si       1          3246
4  masculino urbano       si       1          6123
5  femenino  rural        no       1         11587
6  masculino rural        no       1         10969
7  femenino  urbano       no       1          6134
8  masculino urbano       no       1          6693
9  femenino  rural        si       2           175
10 masculino rural        si       2           136
   round(fitted(mod0), digits = 0)
1   8153                          
2   9493                          
3   4820                          
4   5613                          
5   9971                          
6  11610                          
7   5895                          
8   6864                          
9    118                          
10   137

In [7]:
#prueba de hipotesis para ver si es buen modelo
pchisq(deviance(mod0), df = df.residual(mod0),lower.tail = F)

[1] 0

\Large{Imprimimos los primeros diez datos y sus estimaciones, vemos que la estimación
no es buena. 
La estadítstica de la devianza se distribuye como una $\chi^2$ asi que comparamos con la devianza del modelo, 
$H_o=\text{"Las frecuencias esperadas satisfacen el modelo log lineal dado }"$,
Claramente no lo hace, ya que la prueba anterior se rechaza la hipótesis. 
Se concluye que para este caso no es un buen modelo no considerar las interaciones.}
\normalsize{}

### Modelo Saturado
Continuando con los modelos ahoramos ajustaremos un modelo saturado en el que se consideran todas las posibles
interacciones, como es sabido el ajuste es perfecto pero el número parametros(40) es el mismo que de observaciones.

In [9]:
modSat=glm(Freq ~ genero*localizacion*resultado*cinturon, 
           data=accidentes,family = poisson)

In [10]:
cbind(modSat$data, fitted(modSat))[1:10,]

genero    localizacion cinturon resultado Freq  fitted(modSat)
1  femenino  rural        si       1          7287  7287         
2  masculino rural        si       1         10381 10381         
3  femenino  urbano       si       1          3246  3246         
4  masculino urbano       si       1          6123  6123         
5  femenino  rural        no       1         11587 11587         
6  masculino rural        no       1         10969 10969         
7  femenino  urbano       no       1          6134  6134         
8  masculino urbano       no       1          6693  6693         
9  femenino  rural        si       2           175   175         
10 masculino rural        si       2           136   136

In [11]:
#Hacemos la prueba de Hipótesis y no se rechaza la hupitesis
pchisq(deviance(modSat), df = df.residual(modSat))

[1] 1

### Modelo homogeneo
Ahora vamos a considerar el modelo
\begin{eqnarray}
\log\mu_{ijkl}&=&\log\lambda+\log\lambda_i^X+\log\lambda_j^Y+\log\lambda_k^W+\log\lambda_l^Z+
\log\lambda_i^X\log\lambda_j^Y+\log\lambda_i^X\log\lambda_k^W+\log\lambda_i^X\log\lambda_l^Z+\\
&&\log\lambda_j^Y\log\lambda_k^W+\log\lambda_j^Y\log\lambda_l^Z+\log\lambda_k^W\log\lambda_l^Z
\end{eqnarray}


In [12]:
modHom <- glm(Freq ~ (genero+cinturon+resultado+localizacion)^2, 
              data = accidentes, family = poisson)

In [13]:
FreqEsperada=(round(fitted(mod0), digits = 0))
summary(modHom)
cbind(modHom$data, FreqEsperada)[1:12,]


Call:
glm(formula = Freq ~ (genero + cinturon + resultado + localizacion)^2, 
    family = poisson, data = accidentes)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-2.5099  -0.5499  -0.1179   0.7137   1.9013  

Coefficients:
                                   Estimate Std. Error z value Pr(>|z|)    
(Intercept)                         8.87797    0.01087 816.719  < 2e-16 ***
generomasculino                     0.37911    0.01332  28.454  < 2e-16 ***
cinturonno                          0.49271    0.01311  37.579  < 2e-16 ***
resultado2                         -3.86043    0.06265 -61.616  < 2e-16 ***
resultado3                         -2.28298    0.02946 -77.487  < 2e-16 ***
resultado4                         -4.34789    0.07534 -57.707  < 2e-16 ***
resultado5                         -6.41098    0.20178 -31.772  < 2e-16 ***
localizacionurbano                 -0.76101    0.01603 -47.464  < 2e-16 ***
generomasculino:cinturonno         -0.45933    0.01569 -29.279  < 2

genero    localizacion cinturon resultado Freq  FreqEsperada
1  femenino  rural        si       1          7287  8153       
2  masculino rural        si       1         10381  9493       
3  femenino  urbano       si       1          3246  4820       
4  masculino urbano       si       1          6123  5613       
5  femenino  rural        no       1         11587  9971       
6  masculino rural        no       1         10969 11610       
7  femenino  urbano       no       1          6134  5895       
8  masculino urbano       no       1          6693  6864       
9  femenino  rural        si       2           175   118       
10 masculino rural        si       2           136   137       
11 femenino  urbano       si       2            73    70       
12 masculino urbano       si       2           141    81

In [14]:
#Hacemos la prueba de hipótesis para ver si es un buen modelo,
# pero no lo es, se rechaza la hipotesis. 
pchisq(deviance(modHom), df = df.residual(modHom),lower.tail = F)

[1] 0.000668487

### Modelo con interración a tercer orden  
Ahora vamos a considerar interraciones a tercer nivel es decir

In [15]:
modint3orden <- glm(Freq ~ (genero + localizacion+ resultado + cinturon)^3, 
                    data = accidentes, family = poisson)

In [20]:
FreqEsperada=(round(fitted(modint3orden), digits = 0))
cbind(modint3orden$data, FreqEsperada)[1:12,]

genero    localizacion cinturon resultado Freq  FreqEsperada
1  femenino  rural        si       1          7287  7278       
2  masculino rural        si       1         10381 10390       
3  femenino  urbano       si       1          3246  3255       
4  masculino urbano       si       1          6123  6114       
5  femenino  rural        no       1         11587 11596       
6  masculino rural        no       1         10969 10960       
7  femenino  urbano       no       1          6134  6125       
8  masculino urbano       no       1          6693  6702       
9  femenino  rural        si       2           175   167       
10 masculino rural        si       2           136   144       
11 femenino  urbano       si       2            73    81       
12 masculino urbano       si       2           141   133

In [21]:
#Hacemos la prueba de Hipótesis y no se rechaza la hipotesis nula
pchisq(deviance(modint3orden), df = df.residual(modint3orden))
# El detalle es que tenemos 36 coeficientes práctimente es igual que el 
#numero de observaciones
length(modint3orden$coefficients)

[1] 0.9716035

[1] 36

Sin embargo tenemos 14 coeficientes no significativos, y son las siguientes relaciones 
\begin{center}
\begin{tabular}{ l | l }
generomasculino:resultado5     &        0.417832  \\
localizacionurbano:resultado2      &          0.518781 \\
generomasculino:localizacionurbano:resultado3 &0.456342    \\
generomasculino:localizacionurbano:resultado4 &0.481350    \\
generomasculino:localizacionurbano:resultado5 &0.746419 \\   
generomasculino:resultado2:cinturonno         &0.928027  \\  
generomasculino:resultado3:cinturonno         &0.650300    \\
generomasculino:resultado4:cinturonno         &0.180841    \\
generomasculino:resultado5:cinturonno         &0.198386  \\  
localizacionurbano:resultado2:cinturonno      &0.385063   \\ 
localizacionurbano:resultado3:cinturonno      &0.166413   \\ 
localizacionurbano:resultado4:cinturonno      &0.511436    \\
localizacionurbano:resultado5:cinturonno      &0.982535   \\
resultado5:cinturonno &                    0.010908   
\end{tabular}
\end{center}


Probamos con otras posibles combinaciones para encontrar el mejor modelo. El siguiente como podemos 
observar no es tan buen modelo, en el consideramos el modelo saturado para cinturon, 
localización y genero mas el saturado de cinturon, resultado y localización.

In [22]:
model2 <- glm(Freq ~ (cinturon*localizacion*genero)+(cinturon*resultado*localizacion), data = accidentes, family = poisson)

In [23]:
FreqEsperada=(round(fitted(model2 ), digits = 0))
cbind(model2$data, FreqEsperada)[1:10,]

genero    localizacion cinturon resultado Freq  FreqEsperada
1  femenino  rural        si       1          7287  7514       
2  masculino rural        si       1         10381 10154       
3  femenino  urbano       si       1          3246  3459       
4  masculino urbano       si       1          6123  5910       
5  femenino  rural        no       1         11587 11753       
6  masculino rural        no       1         10969 10803       
7  femenino  urbano       no       1          6134  6270       
8  masculino urbano       no       1          6693  6557       
9  femenino  rural        si       2           175   132       
10 masculino rural        si       2           136   179

In [26]:
### Se rechaza la hpotesis nula 
pchisq(deviance(model2), df = df.residual(model2),lower.tail = F)

[1] 3.785208e-86

El modelo que se muestra a continuación es el modelo anterior más la interacción resultado-genero. Las frecuencias estimadas se muestran a continuación. Tenemos 28 parametros pero 4 de ellos no son significativos.

In [37]:
model3 <- glm(Freq ~ (resultado*genero)+
              (localizacion*cinturon*genero)+(localizacion*cinturon*resultado), 
              data = accidentes, family = poisson)

In [38]:
FreqEsperada=(round(fitted(model3), digits = 0))
cbind(model3$data, FreqEsperada)[1:10,]


genero    localizacion cinturon resultado Freq  FreqEsperada
1  femenino  rural        si       1          7287  7292       
2  masculino rural        si       1         10381 10376       
3  femenino  urbano       si       1          3246  3240       
4  masculino urbano       si       1          6123  6129       
5  femenino  rural        no       1         11587 11608       
6  masculino rural        no       1         10969 10948       
7  femenino  urbano       no       1          6134  6114       
8  masculino urbano       no       1          6693  6713       
9  femenino  rural        si       2           175   156       
10 masculino rural        si       2           136   155

In [39]:
#Hacemos la prueba de Hipótesis y se rechaza la hipotesis nula
pchisq(deviance(model3), df = df.residual(model3),
       lower.tail = F)

[1] 0.02686494

Ahora consieremos el modelo: 

In [40]:
modint4orden <- glm(Freq ~(localizacion+genero+cinturon+resultado)+(localizacion:genero)+
                    (localizacion:cinturon)+(cinturon:resultado)+(localizacion:resultado)+
                        (localizacion:cinturon:genero)+(localizacion:genero:resultado)
                    , data = accidentes, family = poisson)

In [41]:
pchisq(deviance(modint4orden), df = df.residual(modint4orden),
       lower.tail = F)

[1] 0.1221054

In [43]:
FreqEsperada=(round(fitted(modint4orden), digits = 0))
cbind(modint4orden$data, FreqEsperada)[1:10,]

genero    localizacion cinturon resultado Freq  FreqEsperada
1  femenino  rural        si       1          7287  7264       
2  masculino rural        si       1         10381 10384       
3  femenino  urbano       si       1          3246  3265       
4  masculino urbano       si       1          6123  6123       
5  femenino  rural        no       1         11587 11610       
6  masculino rural        no       1         10969 10966       
7  femenino  urbano       no       1          6134  6115       
8  masculino urbano       no       1          6693  6693       
9  femenino  rural        si       2           175   164       
10 masculino rural        si       2           136   141

Considerando las siguientes intereracciones encontramos un buen modelo, ya que al ver las frecuencias estimadas
estas le pegan my bien a las frecuencias observadas. Para prueba de la xi cuadrada no se rechaza la hipótesis. Pero tenemos 28 parametros de los cuales 4 no son significativos. Podemos concluir que el modelo de interacciones a tercer orden 
es más fácil de interpretar y solamente 22 parámetros de los 36 son significativos. 

## Problema 2
En el siguiente problema vamos a ajustar los únicos dos posibles modelos para el caso de dos variables,
usando modelos loglineles nos queda el modelo de independencia y el modelo saturado.
Como podemos ver ninguno de los dos resulta en un buen ajuste, esto es por que los modelos log lineales solamente 
se usan para ver posibles interreccciones entre las categorias de la varibales, sin embargo en este caso la manera correcta de modelar será considerar una variable de respuesta y una explicativa.Para ello usamos el modelo logit.

In [45]:
dataEnf = array(data = c( 4073,2791,624,44,2,2,
                2808, 1693,429,58,14,9,
                879,655,625,280,31,21,
                430,618,1013,123,2,0,
                1102,822,495,210,62,32,
                7,7,9,11,15,17),         
                         
 dim = c(6,6), 
dimnames = list(   
"period" =c("1926-1935","1936-1945","1946-1955","1956-1965",
            "1976-1985","1986-1995"),
"enfermedad"=c("Dip", "Pert", "Tetanus", "Polio", "Measles","Pop") ) )  
tabEnf=ftable(dataEnf, row.vars=c("period"))
#convertimos a data frame para aplicar los analisis
enfermedades<- as.data.frame(as.table(dataEnf))
#imprimimos la tabla
tabEnf
enfermedades[1:12,]                        

          enfermedad  Dip Pert Tetanus Polio Measles  Pop
period                                                   
1926-1935            4073 2808     879   430    1102    7
1936-1945            2791 1693     655   618     822    7
1946-1955             624  429     625  1013     495    9
1956-1965              44   58     280   123     210   11
1976-1985               2   14      31     2      62   15
1986-1995               2    9      21     0      32   17

period    enfermedad Freq
1  1926-1935 Dip        4073
2  1936-1945 Dip        2791
3  1946-1955 Dip         624
4  1956-1965 Dip          44
5  1976-1985 Dip           2
6  1986-1995 Dip           2
7  1926-1935 Pert       2808
8  1936-1945 Pert       1693
9  1946-1955 Pert        429
10 1956-1965 Pert         58
11 1976-1985 Pert         14
12 1986-1995 Pert          9

In [47]:
#modelo saturado
modenfSat0=glm(Freq ~ period*enfermedad, data = enfermedades,
               family = poisson())

In [48]:
FreqEsperada=(round(fitted(modenfSat0), digits = 0))
cbind(modenfSat0$data, FreqEsperada)[1:10,]

period    enfermedad Freq FreqEsperada
1  1926-1935 Dip        4073 4073        
2  1936-1945 Dip        2791 2791        
3  1946-1955 Dip         624  624        
4  1956-1965 Dip          44   44        
5  1976-1985 Dip           2    2        
6  1986-1995 Dip           2    2        
7  1926-1935 Pert       2808 2808        
8  1936-1945 Pert       1693 1693        
9  1946-1955 Pert        429  429        
10 1956-1965 Pert         58   58

In [49]:
#modelo de independencia
modenfIndep=glm(Freq ~ period+enfermedad, data = enfermedades, 
                family = poisson())

In [50]:
FreqEsperada=(round(fitted(modenfIndep), digits = 0))
cbind(modenfIndep$data, FreqEsperada)[1:10,]

period    enfermedad Freq FreqEsperada
1  1926-1935 Dip        4073 3502        
2  1936-1945 Dip        2791 2480        
3  1946-1955 Dip         624 1203        
4  1956-1965 Dip          44  273        
5  1976-1985 Dip           2   47        
6  1986-1995 Dip           2   31        
7  1926-1935 Pert       2808 2328        
8  1936-1945 Pert       1693 1649        
9  1946-1955 Pert        429  800        
10 1956-1965 Pert         58  182

El modelo de independencia estima 1203 cuando para un valor de 624, es un mal modelo. Usando el modelo logit, tenemos:

In [51]:
a=c(4073,2791,624,44,2,2)
b=c(2808, 1693,429,58,14,9)
d=c(879,655,625,280,31,21)
e= c(430,618,1013,123,2,0)
f= c( 1102,822,495,210,62,32)
g=c(7,7,9,11,15,17)
enfermos=cbind(a,b,d,e,f,g)

In [ ]:
periodo<-1:6
cum.logit <- vglm(enfermos ~ periodo, family=cumulative)
summary(cum.logit)
fitted.values(cum.logit)
#ajustando el modelo logit, tenemos las siguentes frecuencias esperadas, las cuales comparadas con 
#las de la tabla teorica podemos notar que obtenemos un buen ajuste para la mayoria de los periodos.

\begin{tabular}{c|c|c|c|c|c|}
4413.13 & 2681.97& 744.83 & 448.12& 1006.98&3.95\\
2256.10 & 1688.216 & 939.95 & 752.66 & 939.95 &9.09\\
738.40 & 568.85 & 626.47& 658.78& 588.17 & 14.31\\
107.28 & 69.20 & 150.37& 222.80 & 165.84 &10.48\\
11.45 & 4.87 & 22.00 & 48.98 & 32.93 & 5.73\\
4.41 & 0.81& 10.10& 34.93 & 19.845& 10.87
\end{tabular}

## Problema 3
#### Solución 

Supongamos que tenemos una muestra  $\{n_{i,j}\}$, con $i=1,\ldots, I$ y $j=1,\ldots, J$ un coteo poisson $\{Y_{i,j}\}$, con $i=1, \ldots, I$, $j=1, \ldots, J$ independientes con medias $\{\mu_{i,j}\}$. La función de verosimilitud es:
\begin{equation*}
l(N; \mu)=\frac{\prod_{i=1}^{I}\prod_{j=1}^{J}e^{-\mu_{i,j}} \mu_{i,j}^{n_{i,j}}}{\prod_{i=1}^{I}\prod_{j=1}^{J}n_{i,j}!},
\end{equation*}
donde $N=\{n_{1,1}, \ldots, n_{I,J}\}$ y $\mu=\{\mu_{1,1}, \ldots, \mu_{I,J}\}$.\\
Si suponemos que $\sum_{i=1}^{n}n_{i,j}=n$ entonces la verosimilitud queda:
\begin{equation*}
l(n; \pi)=\frac{n!}{\prod_{i=1}^{I}\prod_{j=1}^{J}n_{i,j}!}\prod_{i=1}^{I}\prod_{j=1}^{J}\pi_{i,j}^{n_{i,j}},
\end{equation*}
donde $n=\sum_{i=1}^{n}n_{i,j}$ y $\pi=\{\pi_{1,1},\ldots, \pi_{I,J}\}$ y $\pi_{i,j}=\mu_{i,j}/n$.\\
Obtemos que $l(N; \mu)=l(n; \pi)$ siempre que $\sum_{i=1}^{n}n_{i,j}=n$, 
eso quiere decir que tomar una muestra $\{n_{i,j}\}$, con $i=1,\ldots, I$ y $j=1,\ldots, J$  
poisson, la podemos llevar a una muestra observada tipo $\{n_{i,j}\}$  multinomial,  $i=1,\ldots, I$ y $j=1,\ldots, J$ 
con  $\sum_{i=1}^{n}n_{i,j}=n$. Por lo tanto, la distribución Poisson y Multinomial son equivalentes en verosimilitud.

## Problema 4
#### Solución

Debemos encontrar las estadísticas suficientes para 
\begin{equation}
\log(\mu_{i,j})=u+u_{1}(i)+u_{2}(j)+\tau_{i}(v_{j}-\bar{v}).
\end{equation}
como sabemos que la distribución $poisson(\mu)$ pertenece a la familia exponencial, entonces el coeficiente del $\log(\mu)$ es una estadística suficiente. Así que la verosimilitud es
\begin{equation*}
l(N; \mu)=\frac{\prod_{i=1}^{I}\prod_{j=1}^{J}e^{-\mu_{i,j}} \mu_{i,j}^{n_{i,j}}}{\prod_{i=1}^{I}\prod_{j=1}^{J}n_{i,j}!},
\end{equation*}
y donde nos interesa 
\begin{equation}
\log\left(\prod_{i=1}^{I}\prod_{j=1}^{J}e^{-\mu_{i,j}} \mu_{i,j}^{n_{i,j}}\right)=\sum_{i}\sum_{j}n_{i,j}\log(\mu_{i,j})-\sum_{i}\sum_{j}\mu_{i,j}.
\end{equation}
Sustituyendon la primera ecuación en la segunda obtenemos 
\begin{eqnarray*}
\sum_{i}\sum_{j}n_{i,j}[u+u_{1}(i)+u_{2}(j)+\tau_{i}(v_{j}&-&\bar{v})]-\sum_{i}\sum_{j}\mu_{i,j}\\
  =u\sum_{i}\sum_{j}n_{i,j}+\sum_{i}u_{1}(i)\sum_{j}n_{i,j}+\sum_{j}u_{2}(j)\sum_{i}n_{i,j} &+&\sum_{i}\sum_{j}n_{i,j}\tau_{i}(v_{j}-\bar{v})-\sum_{i}\sum_{j}\mu_{i,j}
\end{eqnarray*}
Entonces los coeficientes del modelo dado  son 
\begin{eqnarray*}
\text{el coeficientes de $u$ es }  \sum_{i}\sum_{j}n_{i,j} \text{  el de $u_{1}(i)$  es  }  \sum_{j}n_{i,j}\\
\end{eqnarray*}

\begin{eqnarray*}
\text{el de $u_{2}(j)$ es } \sum_{i}n_{i,j} \text{  y para $\tau_{i}(v_{j}-\bar{v})$ es  } n_{i,j}.
\end{eqnarray*}

y esto para cada $i$ y para cada $j$. Entonces, las estadísticas suficientes están dada por $\left\{\sum_{i}\sum_{j}n_{i,j}, \sum_{j}n_{i,j},  \sum_{i}n_{i,j}, n_{i,j} \right\}$.


## Problema 5
#### Solución

Si definimos $\mu_{i,j}=\mu\alpha_{i}\beta_{j}$, y aplicamos logaritmo, tenemos:
\begin{equation}
\log(\mu_{i,j})=\log(\mu)+\log(\alpha_{i})+\log(\beta_{j}),
\end{equation}
bajo la restricción de que para una misma variable sume cero,
$$
0=\sum_{i}\log(\alpha_{i})=\sum_{h}\log(\alpha_{h})
$$, entonces,
\begin{eqnarray*}
0=\sum_{i}\log(\alpha_{i})-\sum_{h}\log(\alpha_{h})&=&\sum_{i}\log(\alpha_{i})-\sum_{h}\log(\alpha_{h})\frac{I}{I}
=\sum_{i}\left[\log(\alpha_{i})-\frac{\sum_{h}\log(\alpha_{h})}{I}\right].
\end{eqnarray*}
Por lo anteior
\begin{equation}
u_{1}(i)=\log(\alpha_{i})-\frac{\sum_{h}\log(\alpha_{h})}{I} \Rightarrow \sum_{i}u_{1}(i)=0.
\end{equation}
De manera analoga con:
\begin{equation}
u_{2}(j)=\log(\alpha_{j})-\frac{\sum_{h}\log(\beta_{h})}{J} \Rightarrow \sum_{j}u_{2}(j)=0.
\end{equation}
Como el modelo loglineal de independencia está dada por 
\begin{equation*}
\log(\mu_{i,j})=u+u_{1}(i)+u_{2}(j),
\end{equation*}
este último lo podemos escribir como:
\begin{eqnarray*}
\log(\mu_{i,j})&=&\log(\mu)+\log(\alpha_{i})+\log(\beta_{j})+\left(\frac{\sum_{h}\alpha_{h}}{I}+\frac{\sum_{h}\beta_{h}}{J}\right)-\left(\frac{\sum_{h}\alpha_{h}}{I}+\frac{\sum_{h}\beta_{h}}{J}\right)\\
\end{eqnarray*}
hemos sumado un cero, 
\begin{eqnarray*}
\log(\mu_{i,j})&=&\log(\mu)+\left(\frac{\sum_{h}\alpha_{h}}{I}+\frac{\sum_{h}\beta_{h}}{J}\right)+\log(\alpha_{i})-\frac{\sum_{h}\alpha_{h}}{I}+\log(\beta_{j})-\frac{\sum_{h}\beta_{h}}{J}\\
                      &=&u+u_{1}(i)+u_{2}(j) ,
\end{eqnarray*}
donde  $u_{1}(i)=\log(\alpha_{i})-\frac{\sum_{h}\alpha_{h}}{I}$, $u_{2}(j)=\log(\beta_{j})-\frac{\sum_{h}\beta_{h}}{J}$ y $u=\log(\mu)+\frac{\sum_{h}\alpha_{h}}{I}+\frac{\sum_{h}\beta_{h}}{J}$. Se concluye la prueba 


## Problema 6
#### Solución

si definimos $u_{1}(i)=\log(\alpha_{i})-\log(\alpha_{1})$ y $u_{2}(j)=\log(\beta_{j})-\log(\beta_{1})$ entonces para $i=1$
se cumple que 
\begin{eqnarray*}
u_{1}(1)&=&\log(\alpha_{1})-\log(\alpha_{1})=0 \\
u_{2}(1)&=&\log(\beta_{1})-\log(\beta_{1})=0.
\end{eqnarray*}
se cumple la restricción de evaluar un nivel base a cero.
Si sustuimos los valores de $u_{1}(i)$ y $u_{2}(i)$, en 
\begin{eqnarray*}
\log(\mu)+\log(\alpha_{i})+\log(\beta_{j})&=&u+u_{1}(i)+u_{2}(j), 
\end{eqnarray*}
tenemos 
\begin{equation*}
\log(\mu)+\log(\alpha_{i})+\log(\beta_{j})=u+\log(\alpha_{i})-\log(\alpha_{1})+\log(\beta_{j})-\log(\beta_{1}).
\end{equation*}
Simplificando la expresión anterior tenemos:
\begin{eqnarray*}u=\log(\mu)+\log(\alpha_{1})+\log(\beta_{1})
\end{eqnarray*}

## Problema 7
##### Solución

Ahora vamos a ajustar un modelo, simetrico y quasi-simetrico. 

In [2]:
library("gnm")

Warning message:
"package 'gnm' was built under R version 3.3.3"

In [92]:
cafe= array(data=c(93, 9, 17, 6, 10,17, 46, 11, 4, 4,44, 11, 
155, 9, 12, 7, 0, 9, 15,2,10, 9, 12, 2, 27),
dim = c(5,5), 
dimnames = list("primera" = c( "1" ,"2" , "3", "4", "5"),
"segunda"=c( "1" ,"2" , "3", "4", "5")))

cafeTable0=as.data.frame(as.table(cafe))
cafeTable1=ftable(cafe,row.vars=c("primera"))


In [91]:
#Usando la funcion gnm ajustamos un modelo simetrico 
s = gnm(Freq∼Symm(primera,segunda),data=cafeTable0,family=poisson)
summary(s)


Call:
gnm(formula = Freq ~ Symm(primera, segunda), family = poisson, 
    data = cafeTable0)

Deviance Residuals: 
       Min          1Q      Median          3Q         Max  
-2.670e+00  -3.332e-08   0.000e+00   0.000e+00   2.291e+00  

Coefficients:
                         Estimate Std. Error z value Pr(>|z|)    
(Intercept)                4.5326     0.1037  43.711  < 2e-16 ***
Symm(primera, segunda)12  -1.9677     0.2218  -8.870  < 2e-16 ***
Symm(primera, segunda)13  -1.1149     0.1648  -6.767 1.32e-11 ***
Symm(primera, segunda)14  -2.6608     0.2961  -8.986  < 2e-16 ***
Symm(primera, segunda)15  -2.2300     0.2465  -9.047  < 2e-16 ***
Symm(primera, segunda)22  -0.7040     0.1803  -3.905 9.41e-05 ***
Symm(primera, segunda)23  -2.1347     0.2371  -9.004  < 2e-16 ***
Symm(primera, segunda)24  -3.8395     0.5106  -7.519 5.52e-14 ***
Symm(primera, segunda)25  -2.6608     0.2961  -8.986  < 2e-16 ***
Symm(primera, segunda)33   0.5108     0.1312   3.895 9.84e-05 ***
Symm(primera, segunda

In [93]:
print("Tabla original")
cafeTable1
c=as.vector(round(as.data.frame(as.table(fitted.values(s)))[,2], digits = 2))
print("Frecuencias Estimadas")
matrix(c,ncol=5)

[1] "Tabla original"


        segunda   1   2   3   4   5
primera                            
1                93  17  44   7  10
2                 9  46  11   0   9
3                17  11 155   9  12
4                 6   4   9  15   2
5                10   4  12   2  27

[1] "Frecuencias Estimadas"


[,1] [,2] [,3]  [,4] [,5]
[1,] 93.0 13.0  30.5  6.5 10.0
[2,] 13.0 46.0  11.0  2.0  6.5
[3,] 30.5 11.0 155.0  9.0 12.0
[4,]  6.5  2.0   9.0 15.0  2.0
[5,] 10.0  6.5  12.0  2.0 27.0

In [95]:
QS = gnm(Freq∼Symm(primera,segunda)+primera,data=cafeTable0,family=poisson)

In [96]:
print("Tabla original")
cafeTable1
c=as.vector(round(as.data.frame(as.table(fitted.values(QS)))[,2], digits = 2))
print("Frecuencias Estimadas")
matrix(c,ncol=5)

[1] "Tabla original"


        segunda   1   2   3   4   5
primera                            
1                93  17  44   7  10
2                 9  46  11   0   9
3                17  11 155   9  12
4                 6   4   9  15   2
5                10   4  12   2  27

[1] "Frecuencias Estimadas"


[,1]  [,2]  [,3]   [,4]  [,5] 
[1,] 93.00 16.78  40.88  7.45 12.89
[2,]  9.22 46.00  11.60  1.70  6.49
[3,] 20.12 10.40 155.00  7.16 11.32
[4,]  5.55  2.30  10.84 15.00  2.30
[5,]  7.11  6.51  12.68  1.70 27.00

In [98]:
summary(QS)


Call:
gnm(formula = Freq ~ Symm(primera, segunda) + primera, family = poisson, 
    data = cafeTable0)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.8419  -0.1925   0.0000   0.1999   1.0215  

Coefficients:
                         Estimate Std. Error z value Pr(>|z|)    
(Intercept)                4.5326     0.1037  43.711  < 2e-16 ***
Symm(primera, segunda)12  -1.7122     0.2433  -7.036 1.98e-12 ***
Symm(primera, segunda)13  -0.8220     0.1799  -4.568 4.92e-06 ***
Symm(primera, segunda)14  -2.5249     0.3319  -7.608 2.78e-14 ***
Symm(primera, segunda)15  -1.9760     0.2677  -7.382 1.56e-13 ***
Symm(primera, segunda)22  -0.1045     0.3348  -0.312 0.754932    
Symm(primera, segunda)23  -1.4821     0.3179  -4.662 3.14e-06 ***
Symm(primera, segunda)24  -3.4047     0.5716  -5.957 2.57e-09 ***
Symm(primera, segunda)25  -2.0634     0.3788  -5.447 5.12e-08 ***
Symm(primera, segunda)33   1.2196     0.2556   4.772 1.83e-06 ***
Symm(primera, segunda)34  -1.8558     0.3